chore: publish a runfiles library as a wheel

alexeagle opened the pull request:
Wire it up to GH actions so it is published for each release.

Tested locally with:
```
$ bazel build python/runfiles:wheel --embed_label=1.0.2 --stamp
$ PYTHONPATH=bazel-bin/python/runfiles/bazel_runfiles-_BUILD_EMBED_LABEL_-py3-none-any.whl python
>>> import runfiles
>>> runfiles.Create()
```

Note, I would have liked to call the package bazel-runfiles, but this isn't possible without either refactoring the paths in this repo, or doing some fancy starlark to copy files around to create a folder that we turn into the wheel. There is no project https://pypi.org/project/runfiles though there is a https://pypi.org/project/runfile

We could try harder to get the name we prefer.

alexeagle commented:
Looks like our buildifier presubmit check is broken, it complains about `MODULE.bazel` which I didn't edit in this PR.

```
##### :bazel: buildifier: found 1 format issue in your WORKSPACE, BUILD and *.bzl files
--
  | Please download <a href="https://github.com/bazelbuild/buildtools/releases/tag/6.0.0">buildifier 6.0.0</a> and run the following command in your workspace:<br/><pre><code>buildifier MODULE.bazel</code></pre>
```

fmeum commented:
It's broken because the new buildifier release formats `MODULE.bazel` files differently. You either need to pin to the old release or apply the fix.

alexeagle commented:
Yeah, I rebased over a commit I added on `main` that updated that formatting, and yeah I knew about your buildifier format fix. It feels like a bug to me that our presubmit.yml uses "latest" of buildifier but not a fight I want to pick today :)

fmeum commented:
The logic in `CurrentRepository` makes some assumptions about the runfiles path of the file that contains it. We probably have to add tests for it when loaded from the wheel.

alexeagle commented:
Yeah I'll try adding the integration test before merging this.

rickeylev reviewed (approved):
overall LGTM.
+1 to having some sort of test; I'm ok with that being a separate change though, too

alexeagle commented:
@rickeylev looks like your personal email isn't properly connected with your GitHub identity in a way that Google's CLA bot understands - this is the result of applying your suggestions. You might need to add that email address in http://go/github ?

rickeylev commented:
CLA issue fixed

alexeagle commented:
Cool, thanks!
